### PR TITLE
Fix for issue #38780

### PR DESCRIPTION
**Pull request by welovej**
# Contributing to the Azure SDK

Please see our [CONTRIBUTING.md](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md) if you are not familiar with contributing to this repository or have questions.

For specific information about pull request etiquette and best practices, see [this section](https://github.com/Azure/azure-sdk-for-net/blob/main/CONTRIBUTING.md#pull-request-etiquette-and-best-practices).

### CODE REVIEW

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs [hunk 1]
```diff
@@ -0,0 +1,39 @@
...
+        private static void DeserializeTypeValue(JsonProperty property, ref ResourceType type)
+        {
+            try
+            {
+                type = new ResourceType(property.Value.GetString());
+            }
+            catch (Exception)
+            {
+                // type always is null, https://github.com/Azure/azure-sdk-for-net/issues/38780, so we need to handle the exception
+                type = default;
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:35]: The type here should be `ApplicationInsightsWorkbookResource.ResourceType`.

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs [hunk 2]
```diff
@@ -0,0 +1,39 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+// <auto-generated/>
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:4]: remove this line

**Review by ArthurMa1978 — CHANGES_REQUESTED** [review]
Add test for this fix

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs (hunk 1 above)
Comment (ArcturusZhang) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:35]: it is unfortunate that ResourceType does not provide a method `TryParse` and here we have to rely on exceptions.
I think we need to track this issue.

Comment (ArcturusZhang) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:35]: Created an issue: https://github.com/Azure/azure-sdk-for-net/issues/45273

Comment (welovej) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:35]: Fixed

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs (hunk 2 above)
Comment (welovej) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:4]: Fixed

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+                    Kind = WorkbookSharedTypeKind.Shared,
+                    Tags =
+                  {
+                ["TagSample01"] = "sample01",
+                ["TagSample02"] = "sample02",
+                  },
+                };
+                string sourceId = $"/subscriptions/{subscriptionId}/resourcegroups/{resourceGroupName}";
+                ArmOperation<ApplicationInsightsWorkbookResource> lro = await collection.CreateOrUpdateAsync(WaitUntil.Completed, resourceName, data, sourceId);
+                ApplicationInsightsWorkbookResource result = lro.Value;
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs:53]: something you can assert here?

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs [hunk 4]
```diff
@@ -0,0 +1,26 @@
...
+    /// A workbook definition.
+    /// </summary>
+    [CodeGenSerialization(nameof(ResourceType), DeserializationValueHook = nameof(DeserializeTypeValue))]
+    public partial class ApplicationInsightsWorkbookData : TrackedResourceData
+    {
+        [MethodImpl(MethodImplOptions.AggressiveInlining)]
+        private static void DeserializeTypeValue(JsonProperty property, ref ResourceType type)
+        {
+            var propVal = property.Value.GetString();
+            type = string.IsNullOrEmpty(propVal) ? ApplicationInsightsWorkbookResource.ResourceType : new ResourceType(propVal);
```
Comment (ArcturusZhang) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/src/Customization/ApplicationInsightsWorkbookData.cs:23]: should it be just empty when the string is null or empty?
```suggestion
            type = string.IsNullOrEmpty(propVal) ? default : new ResourceType(propVal);
```

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs (hunk 3 above)
Comment (ArthurMa1978) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs:53]: like check the resourcetype

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs [hunk 5]
```diff
@@ -0,0 +1,62 @@
...
+            }
+            catch (Exception e)
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs:55]: don't try ... catch in this test which swallow errors

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/CHANGELOG.md [hunk 6]
```diff
@@ -7,7 +7,7 @@
 ### Breaking Changes
 
 ### Bugs Fixed
-
+- Fixed 'CreateOrUpdateAsync's InvalidResourceType problem for ApplicationInsightsWorkbookResource, Issue at https://github.com/Azure/azure-sdk-for-net/issues/38780
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/CHANGELOG.md:10]: add new line before this line and after this line

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs [hunk 7]
```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
```
Comment (ArthurMa1978) [inline sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs:2]: add new line after this

File: sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs (hunk 3 above)
Comment (welovej) [reply sdk/applicationinsights/Azure.ResourceManager.ApplicationInsights/tests/Scenario/ApplicationInsightsWorkbookResource_Test.cs]: Fixed

**Review by ArthurMa1978 — APPROVED** [review]

**Pull request merged**